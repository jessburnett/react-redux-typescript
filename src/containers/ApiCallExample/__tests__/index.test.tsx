import * as React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { ApiCallExample, mapStateToProps, mapDispatchToProps } from '../';

const testData = [1,2,3];

describe('<ApiCallExample />', () => {

  describe('render correctly', () => {

    it('', () => {
      expect(<ApiCallExample />).toMatchSnapshot();
    });

    it('with loading indicator', () => {
      expect(<ApiCallExample loading={true} />).toMatchSnapshot();
    });

    it('with error', () => {
      expect(<ApiCallExample error={true} />).toMatchSnapshot();
    });

    it('with data', () => {
      expect(<ApiCallExample data={testData} />).toMatchSnapshot();
    });

  });

  it('calls onButtonClick after button is clicked', () => {
    const onButtonClick = jest.fn();
    const wrapper = shallow(<ApiCallExample data={fromJS(testData)} onButtonClick={onButtonClick} />);
    const button = wrapper.find('button').simulate('click');
    expect(onButtonClick).toHaveBeenCalledTimes(1);
  });

  it('maps state to props correctly', () => {
    const state = fromJS({
      apiCallExample: {
        data: testData,
        error: false,
        loading: false,
      },
    });

    expect(mapStateToProps(state)).toMatchSnapshot();
  });

  describe('maps dispatch to props correctly', () => {
    const dispatch = jest.fn();
    let tempProps;

    beforeEach(() => {
      dispatch.mockClear();
      tempProps = mapDispatchToProps(dispatch);
    });

    it('onButtonClick prop', () => {
      tempProps.onButtonClick();
      expect(dispatch.mock.calls[0]).toMatchSnapshot();
    });

  });

});
