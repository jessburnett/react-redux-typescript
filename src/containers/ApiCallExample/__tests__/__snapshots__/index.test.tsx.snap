exports[`<ApiCallExample /> maps dispatch to props correctly onButtonClick prop 1`] = `
Array [
  Object {
    "type": "GET_DATA",
  },
]
`;

exports[`<ApiCallExample /> maps state to props correctly 1`] = `
Object {
  "data": Array [
    1,
    2,
    3,
  ],
  "error": false,
  "loading": false,
}
`;

exports[`<ApiCallExample /> render correctly  1`] = `<ApiCallExample />`;

exports[`<ApiCallExample /> render correctly with data 1`] = `
<ApiCallExample
  data={
    Array [
      1,
      2,
      3,
    ]
  } />
`;

exports[`<ApiCallExample /> render correctly with error 1`] = `
<ApiCallExample
  error={true} />
`;

exports[`<ApiCallExample /> render correctly with loading indicator 1`] = `
<ApiCallExample
  loading={true} />
`;
